### PR TITLE
Include type definitions in build

### PR DESCRIPTION
**Pull request by anthonynichols**
Fixes https://github.com/Koenkk/zigbee-herdsman/issues/121

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -73,7 +74,7 @@
     "build": "./node_modules/.bin/tsc",
     "start": "./node_modules/.bin/tsc -w",
     "test": "jest test",
-    "test-with-coverage": "jest test --coverage --collectCoverageFrom '**/*.{ts,tsx}'",
+    "test-with-coverage": "jest test --coverage --collectCoverageFrom 'src/**/*.ts'",
```
Comment (anthonynichols) [inline package.json]: `test-with-coverage` was failing for me, and I don't think there are any React `.tsx` files, so I changed this to only look at `.ts` files in the `src` directory.

File: tsconfig.json [hunk 2]
```diff
@@ -6,18 +6,14 @@
         "noImplicitAny": true,
         "moduleResolution": "node",
         "sourceMap": true,
+        "declaration": true,
+        "declarationMap": true,
         "outDir": "dist",
         "baseUrl": ".",
-        "allowJs": true,
```
Comment (anthonynichols) [inline tsconfig.json]: There aren't any `.js` files being transpiled by TypeScript, so this can be removed.

File: tsconfig.json [hunk 3]
```diff
@@ -6,18 +6,14 @@
...
         "resolveJsonModule": true,
-        "paths": {
-            "*": [
-                "node_modules/*"
-            ]
-        }
```
Comment (anthonynichols) [inline tsconfig.json]: This is unnecessary as the default module resolution strategy for `node` is to look first in `node_modules/*`.

**Comment by Koenkk** [comment]
Thanks for all the improvements!

**Pull request merged**